consult.el

```diff
@@ -665,15 +665,20 @@ Otherwise replace the just-yanked text with the selected text."
 
 (defun consult--register-candidates ()
   "Return alist of register descriptions and register names."
-  (mapcar
-   (lambda (r)
-     (setq r (car r))
-     (cons (format "%s: %s"
-                   (single-key-description r)
-                   (register-describe-oneline r))
-           r))
-   (or (sort (copy-sequence register-alist) #'car-less-than-car)
-       (user-error "All registers are empty"))))
+  (cl-remove-if
+   #'null
+   (mapcar
+    (lambda (r)
+      ;; Sometimes, registers are made without a `cdr'.  They don't do anything,
+      ;; and should be ignored.
+      (when (cdr r)
```

Comment:
Please use seq-remove instead of cl- and filter before the mapcar. Then we can avoid the when here.

Reply:
How is it now?

Also, what is the benefit of using `seq-remove/filter` over the `cl-lib` versions?

Reply:
Only stylistic and for consistency. I think seq can be treated as more fundamental since seq is somehow Emacs stdlib. I am personally not a fan of overusing the cl lib. I am using only a few things like cl-defun for kw args. You might have noticed that I replaced all uses of cl-loop with other iteration constructs. I like your loopy more than cl. But even more preferable to such dsls is composing functions. I am preferring the functional style maybe combined with a bit of mutation. But I am also a haskeller. In cases where composing multiple functions is not good enough for performance I am ok with imperative style dolist or cl-loop if it looks nicer.